bgpd/bgp_rpki.c

```diff
@@ -902,9 +903,8 @@ static void free_cache(struct cache *cache)
 	if (cache->type == TCP) {
 		XFREE(MTYPE_BGP_RPKI_CACHE, cache->tr_config.tcp_config->host);
 		XFREE(MTYPE_BGP_RPKI_CACHE, cache->tr_config.tcp_config->port);
-		if (cache->tr_config.tcp_config->bindaddr)
-			XFREE(MTYPE_BGP_RPKI_CACHE,
-			      cache->tr_config.tcp_config->bindaddr);
+		XFREE(MTYPE_BGP_RPKI_CACHE,
```

Comment:
Why remove the if(...) check? Is it guaranteed that cache->tr_config.tcp_config->bindaddr will always be non-NULL?